app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt

```diff
@@ -0,0 +1,572 @@
+import org.oppia.android.util.threading.BackgroundDispatcher
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [FeatureFlagsFragment]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = FeatureFlagsFragmentTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class FeatureFlagsFragmentTest {
+
+  private companion object {
+    private const val DATABASE_NAME = "platform_parameter_and_feature_flag_database"
+    private const val DOWNLOADS_SUPPORT_FLAG_NAME = "Downloads Support"
+  }
+
+  @get:Rule val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+  @get:Rule val oppiaTestRule = OppiaTestRule()
+  @Inject lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+  @Inject lateinit var platformParameterControllerDebugImpl: PlatformParameterControllerDebugImpl
+  @Inject lateinit var monitorFactory: DataProviderTestMonitor.Factory
+  @Inject lateinit var context: Context
+
+  @Test
+  fun testFeatureFlagsFragment_verifyRecyclerView_hasCorrectItemCount() {
+    setUpTestApplicationComponent()
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      val expectedCount = getEphemeralFeatureFlags().size
+
+      onView(withId(R.id.feature_flags_recycler_view))
+        .check(RecyclerViewMatcher.hasItemCount(count = expectedCount))
+
+      // Note to developers: if you add/remove a feature flag, please update the expected count.
+      onView(withId(R.id.feature_flags_recycler_view))
+        .check(RecyclerViewMatcher.hasItemCount(count = 14))
+    }
+  }
+
+  @Test
+  fun testFeatureFlagsFragment_verifyRecyclerViewItems_hasCorrectDetails() {
+    setUpTestApplicationComponent()
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      getEphemeralFeatureFlags().forEachIndexed { index, ephemeralFeatureFlag ->
+        scrollToPosition(index)
+        verifyFeatureFlagDisplayName(
+          index,
+          getFeatureFlagDisplayName(ephemeralFeatureFlag.id)
+        )
+        verifyFeatureFlagSyncStatus(
+          index,
+          getSyncStatusText(ephemeralFeatureFlag.syncStatus)
+        )
+        verifyFeatureFlagSwitchState(
+          index,
+          ephemeralFeatureFlag.currentValue
+        )
+        verifyFeatureFlagBackgroundColor(
+          index,
+          when (ephemeralFeatureFlag.syncStatus) {
+            SyncStatus.SYNCED_FROM_SERVER -> {
+              0xFF00645C.toInt()
+            }
+            SyncStatus.NOT_SYNCED_FROM_SERVER -> {
+              0xFFBE563C.toInt()
+            }
+            else -> 0xFF00645C.toInt()
+          }
+        )
+      }
+    }
+  }
+
+  @Test
+  fun testFeatureFlagFragment_withNoRemoteOrOverridenDownloadsSupportFlag_hasCorrectDetails() {
```

Comment:
Here and below - ref: https://github.com/oppia/oppia-android/pull/5861/files/027ba66557c4a435c730a129ecb4f0d6cdc97c3f#r2160131490
```suggestion
  fun testFeatureFlagsFragment_withNoRemoteOrOverridenDownloadsSupportFlag_hasCorrectDetails() {
```

Reply:
updated